File: game_server/ServerProtocol/src/Common/ChapterEnum/EChapterEventType.java
package Common.ChapterEnum;

/*********
 * 关卡事件类型
 **/
public enum EChapterEventType {
	NONE, //0 ==== 
	REWARD, //1 ==== 奖励
	DISPATCH, //2 ==== 大臣派遣
	CHOICE, //3 ==== 选择
	;
public static final EChapterEventType[]  EChapterEventType_Values = EChapterEventType.values();
public static final int EChapterEventType_Length = EChapterEventType_Values.length;
public static EChapterEventType EChapterEventType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= EChapterEventType_Length){ return null; }
	return EChapterEventType_Values[_ivalue];
}
}

File: game_client/Assets/Scripts/ClientProtocol/Common/ChapterEnum/EChapterEventType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.ChapterEnum
{

/// <summary>
/// 关卡事件类型
/// </summary>
public enum EChapterEventType {
	NONE, //0 ==== 
	[InspectorName("REWARD - idx[1] - 奖励")]
	REWARD, //1 ==== 奖励
	[InspectorName("DISPATCH - idx[2] - 大臣派遣")]
	DISPATCH, //2 ==== 大臣派遣
	[InspectorName("CHOICE - idx[3] - 选择")]
	CHOICE, //3 ==== 选择
}

public class EChapterEventTypeComparer : IEqualityComparer<EChapterEventType>{
	public bool Equals(EChapterEventType x, EChapterEventType y) { return x == y; }
	public int GetHashCode(EChapterEventType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 4;
}
}

